src/platform/android/DeviceInstanceInfoProviderImpl.cpp

```diff
@@ -73,6 +73,22 @@ CHIP_ERROR DeviceInstanceInfoProviderImpl::GetProductLabel(char * buf, size_t bu
     return Internal::AndroidConfig::ReadConfigValueStr(Internal::AndroidConfig::kConfigKey_ProductLabel, buf, bufSize, dateLen);
 }
 
+CHIP_ERROR DeviceInstanceInfoProviderImpl::GetDeviceName(MutableCharSpan & deviceNameSpan)
+{
+    CHIP_ERROR err;
+    size_t deviceNameSize = 0;
+    err = Internal::AndroidConfig::ReadConfigValueStr(Internal::AndroidConfig::kConfigKey_MfrDeviceName, deviceNameSpan.data(),
+                                                      deviceNameSpan.size(), deviceNameSize);
+    if (err == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND)
```

Comment:
But if this succeeds, what's setting the span size?

Stopped here.  Please go through these implementations carefully and make sure they actually follow the contract for a MutableCharSpan outparam.

Reply:
set span size